File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;

        int maxc = 0;
        int maxn = -1;
        for(int i =0; i < n; i++){
        int count = 0;
            for(int j = 0; j < n; j++){
                if(nums[i] == nums[j]){
                    count++;
                }
                if(count > maxc){
                    maxc = count;
                    maxn = i;
                }
            }
        }
        if(maxc > n/2){
        return nums[maxn];
            }
            return 0;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int i = 0;
        int cnt = 0;
        int n = nums.size();

        int el = nums[i];
        while(i < n){
            if(nums[i] == el) cnt++;
            else cnt--;

            if(cnt == 0) el = nums[i+1];
            i++;
        }
        // cnt = 0;
        // for(auto it : nums){
        //     if(it == el) cnt++;
        // }

        // if(cnt > n/2) return el;
        // return -1;
        return el;
    }
};